### ISSUE TITLE
Session doesn't start when cosmic-greeter is the display manager

### ISSUE THREAD

**Issue by git-f0x** [pop-os/cosmic-greeter#8]
Seems like some permission issue with `cosmic-greeter` causes `cosmic-comp` to panic (`cosmic-greeter` is never displayed, just the message in lines 116-132 in the attached file repeatedly being refreshed). 
Output of `journalctl -e`: 
[journalctl_greeter.txt](https://github.com/pop-os/cosmic-greeter/files/14187151/journalctl_greeter.txt)

I started `cosmic-session` from a different tty at the end and it starts up fine.
Sorry if it is a bit early to report an issue on something just implemented.

**Comment by jackpot51** [comment]
You'll probably need to try again with the latest master branch commit.

**Comment by git-f0x** [comment]
I'll test it as soon as it's pushed to apt.

**Comment by jackpot51** [comment]
It looks like something went wrong with an older version of the package, can you remove the cosmic-greeter user and then reinstall the package?

**Comment by git-f0x** [comment]
I ran `sudo userdel -f cosmic-greeter` then `sudo apt reinstall cosmic-greeter`, but it still just boots to a black screen. Should there have been a reboot between deleting the user and reinstalling?

**Comment by jackpot51** [comment]
what is `grep cosmic-greeter /etc/passwd`?

On my system, it shows:

```
cosmic-greeter:x:131:140:COSMIC Greeter:/var/lib/cosmic-greeter:/bin/false
```

In the newest logs you sent, it looks like the home directory may be /home/cosmic-greeter instead of /var/lib/cosmic-greeter

**Comment by git-f0x** [comment]
`cosmic-greeter:x:121:137:COSMIC Greeter:/var/lib/cosmic-greeter:/bin/false`
This is the log after:
[journalctl_greeter.txt](https://github.com/pop-os/cosmic-greeter/files/14197224/journalctl_greeter.txt)

**Comment by jackpot51** [comment]
Does /var/lib/cosmic-greeter exist, and is it owned by cosmic-greeter?

**Comment by git-f0x** [comment]
```
~$ stat /var/lib/cosmic-greeter
  File: /var/lib/cosmic-greeter
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: 813h/2067d	Inode: 1183148     Links: 2
Access: (0755/drwxr-xr-x)  Uid: (  998/ UNKNOWN)   Gid: (  998/ UNKNOWN)
Access: 2024-02-06 20:30:35.293792404 +0100
Modify: 2024-02-06 20:30:35.293792404 +0100
Change: 2024-02-06 20:30:35.293792404 +0100
 Birth: 2024-02-06 20:30:35.293792404 +0100
```
Seems to be an unknown user.

**Comment by jackpot51** [comment]
What GPU do you have?

**Comment by XV-02** [comment]
I can second issues with using `cosmic-greeter` for session locking after updates since Feb 5th. I the system seems non-responsive after I lock it, and I cannot unlock it (still need to spin up a lab unit and pull logs around this)

I would also note it may not be directly related to Cosmic sessions, and that Pop!_OS Wayland-Gnome sessions are also not launching for me with `cosmic-greeter`. I _can_ launch Xorg sessions ([with caveats](https://github.com/pop-os/cosmic-greeter/issues/9)) though. This may point to a general issue launching Wayland sessions, and not a specific issue launching just `cosmic-session`. I'd like to see how `cosmic-greeter` handles other wayland sessions, so that's next on my list.

**Comment by git-f0x** [comment]
I've reinstalled Pop!_OS (installed in only 43 seconds!), and` cosmic-greeter` now shows up and can log in to `cosmic-session`.
Though if I change the chosen session (e.g. from COSMIC to Pop/Pop on Wayland), the greeter stops responding to pressing **Enter**, even if the chosen session is switched back to COSMIC, and the reboot and power off buttons don't work, so I have to switch to a different tty to reboot.

Locking with `Super + Escape` (the button in `cosmic-applet-power` doesn't do anything) and unlocking works for me .

Should this be closed, since the issue was caused by something on my end?
